src/data/nat/factorization.lean

```diff
@@ -385,6 +385,82 @@ begin
   exact support_add_eq (factorization_disjoint_of_coprime hab),
 end
 
+/-! ### Factorization and `is_square` -/
+
+lemma is_square_iff_factorization_even {m : ℕ} :
+  is_square m ↔ ∀ p, even (m.factorization p) :=
+begin
+  rcases eq_or_ne m 0 with rfl | hm_zero,
+  { simp, },
+  { split,
```

Comment:
It is acceptable to skip the indentation after you got rid of a trivial goal.
```suggestion
  split,
```
(and the rest of the proof)

Reply:
Done!